Automattic/jetpack
# ci: Try faking a connection in plugin update test

## Proposed changes:

We've had some cases where problematic code didn't trigger during the test because it needs the site to be connected.

PR #35911 suggests a way to fake the connection. We can't really use that directly since we want the code for all plugins, not just Jetpack-the-plugin, but we can copy-paste the relevant bit of code into the test script.

Also, while we're at it, let's run each test with a clean copy of the database.

### Other information:


## Jetpack product discussion


p1705092068241859-slack-CBG1CP4EN

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* I think you'd have to merge this in a fork, set up the right secrets so the workflow can run, then make a PR that will break upgrades (like what #34739 did maybe) and see if the CI detects it.
